ImageLounge/src/DkCore/DkImageStorage.cpp

```diff
@@ -280,6 +287,86 @@ QImage DkImage::rotateImage(const QImage &img, double angle)
     return imgR;
 }
 
+QImage transposeImage(const QImage &img)
+{
+    QImage imgIn = img;
+    if (img.depth() > 32) {
+        // Ensure the data is 32-bit aligned
+        imgIn = img.convertToFormat(QImage::Format_ARGB32);
+    }
+
+    QImage imgOut = QImage(imgIn.size().transposed(), imgIn.format());
+
+    const QRgb *dataIn = reinterpret_cast<const QRgb *>(imgIn.constBits());
+    QRgb *dataOut = reinterpret_cast<QRgb *>(imgOut.bits());
+
+    const int h = imgIn.height();
+    const int w = imgIn.width();
+    for (int i = 0; i < h; i++) {
+        for (int j = 0; j < w; j++) {
+            dataOut[j * h + i] = dataIn[i * w + j];
+        }
+    }
+
+    return imgOut;
+}
+
+void flipImageHor(QImage &img)
+{
+    if (img.depth() > 32) {
+        // Ensure the data is 32-bit aligned
+        img = img.convertToFormat(QImage::Format_ARGB32);
+    }
+
+    QRgb *data = reinterpret_cast<QRgb *>(img.bits());
+    const int h = img.height();
+    const int w = img.width();
+    for (int i = 0; i < h; i++) {
+        std::reverse(data + w * i, data + w * (i + 1));
+    }
+}
+
+QImage rotateImageFast(const QImage &img, double angle)
+{
+    angle = std::fmod(angle, 360);
+    if (angle < 0) {
+        angle += 360;
+    }
+
+    if (angle == 0) {
+        return img;
+    }
+
+    if (angle == 90 || angle == 180 || angle == 270) {
```

Comment:
You might consider QImage::mirrored() if you are going to stack operations anyways.

Reply:
I tried this, and it's actually slightly faster. It can work for 90 and 270 degrees.